docs/maintenance/recompress-storage.md

```diff
@@ -0,0 +1,202 @@
+storage layer.  Among better performance, the changes result in
+*significant* reduction of the storage on archival nodes: from 7.9 TB
+to 3.1 TB on mainnet and from 4.3 TB to 1.5 TB on testnet.
+
+However, to take full advantage of the saving some manual steps need
+to be taken.  This document describes those steps.  Note that if
+you’re not running an archival node, you don’t need to do anything.
+
+One of the improvements is changing of the compression algorithm used
+by the databases.  Because of that, the whole process is referred to
+as ‘recompressing the storage’.
+
+
+## Performing the operation
+
+The operation can be performed at any time.  That is, it does not need
+to happen directly after the node upgrade.  In fact, it might be
+better to schedule it some number of days after binary upgrade.  Such
+delay gives time to make sure that the binary release works correctly
+before introducing another change.
+
+### Required free space
+
+Recompressing storage works by creating *a new* database.  As
+a result, it requires noticeable amount of free disk space.
+Specifically 4 TB for mainnet and 2 TB for testnet archival nodes.
+
+If your host does not have enough free space you will need to resize
+one of the existing partitions (if you are running in the cloud and
+can resize disks freely) or temporarily attach a new disk.
+
+At the end of the recompression, once the new storage is created, it
+will replace the old one.  The old database can be kept as a backup or
+deleted to free space it uses.
+
+### Stopping the node
+
+While the recompression takes place the database cannot be modified.
+In other words, the node needs to be shut down.  The operation takes
+around twelve hours on mainnet and six hours on testnet.  If you
+require a continuous access to an archival node, you will need to set
+up a redundant node if one is not already available.
+
+Procedure for stopping a node depends on configuration of your system.
+For example, if nerad is run as a systemd service, the command to stop
+it might look like:
+
+```console
+$ sudo systemctl stop neard.service
+```
+
+### Executing the recompression
+
+The recompression is done by executing `recompress-storage` command of
+`neard` binary.  If the command is interrupted the whole operation
+needs to be run from scratch.  Because of that, it’s best to run it
+inside of a screen or tmux session or through nohup utility.  For
+example:
+
+```console
+$ NEAR_HOME=$HOME/.near  # change to point to near home directory
+$ export NEAR_HOME
+
+$ screen
+$ # Inside of screen session:
+$ neard recompress-storage --output-dir="${NEAR_HOME:?}/data.new"
+```
+
+Once recompression finishes successfully, the new database needs to be
+put in place of the old one.  This is simply done by renaming the
+`data.new` directory:
+
+```console
+$ NEAR_HOME=$HOME/.near  # change to point to near home directory
+$ export NEAR_HOME
+
+$ mv -- "${NEAR_HOME:?}/data"     "${NEAR_HOME:?}/data.bak"
```

Comment:
What's the purpose of keeping the non-compressed data as a backup instead of deleting it?

Reply:
None really other than just having a backup.  I’ve extended the section with paragraph talking about this.